plugins/reportal/src/azkaban/reportal/util/ReportalUtil.java

```diff
@@ -57,4 +62,63 @@ public static IStreamProvider getStreamProvider(String fileSystem) {
 
     return sortedNodes;
   }
+
+  /**
+   * Get runtime variables to be set in unscheduled mode of execution.
+   * Returns empty list, if no runtime variable is found
+   *
+   * @param variables
+   * @return
+   */
+  public static List<Variable> getRunTimeVariables(
+    Collection<Variable> variables) {
+    List<Variable> runtimeVariables =
+      ReportalUtil.getVariablesByRegex(variables, "(?!(^"
+        + Reportal.REPORTAL_CONFIG_PREFIX + ")).+");
+
+    return runtimeVariables;
+  }
+
+  /**
+   * Shortlist variables which match a given regex. Returns empty props, if no
+   * eligible variable is found
+   *
+   * @param variables
+   * @param regex
+   * @return
+   */
+  public static List<Variable> getVariablesByRegex(
```

Comment:
what is the expected behavior of this function if 
variables!=null && regex==null
?

Should cover it in the javadoc and/or code